### PR TITLE
[ISC-15184] Adds skipped packet logging

### PR DESCRIPTION
**Pull request by zainkabani**

### CODE REVIEW

File: src/logging.c [hunk 1]
```diff
@@ -65,6 +65,7 @@ static uint16_t file_id = 0;
 
 static const char *reload_command = "RELOAD";
 static const char connect_char = '!';
+static const char skipped_char = 'S';
```
Comment (felipethecarrot) [inline src/logging.c]: Ditto on the other review

https://github.com/instacart/go-replayer/pull/38

File: src/logging.c [hunk 2]
```diff
@@ -210,6 +211,44 @@ void log_connect_to_buffer(bool connected, PgSocket *client) {
...
+
+  /* custom type */
+  memset(buf + len, skipped_char, sizeof(char));
+  len += sizeof(char);
+
+  /* length of the reload command */
+  memcpy(buf + len, &net_skipped_len, sizeof(uint32_t));
+  len += sizeof(uint32_t);
+
+  /* reload itself */
```
Comment (felipethecarrot) [inline src/logging.c]: This comment does not match the behavior

File: src/logging.c [hunk 3]
```diff
@@ -210,6 +211,44 @@ void log_connect_to_buffer(bool connected, PgSocket *client) {
...
+  /* skipped packet length */
+  memcpy(buf + len, &net_pkt_len, sizeof(net_pkt_len));
+  // log_info("Interval %d", buf[len]);
+  len += sizeof(net_pkt_len);
+
+  /* custom type */
+  memset(buf + len, skipped_char, sizeof(char));
+  len += sizeof(char);
+
+  /* length of the reload command */
```
Comment (felipethecarrot) [inline src/logging.c]: This comment does not match the behavior

File: src/logging.c [hunk 4]
```diff
@@ -210,6 +211,44 @@ void log_connect_to_buffer(bool connected, PgSocket *client) {
   len += sizeof(uint8_t);
 }
 
+
+/*
+ * Log skipped packet to buffer.
+ */
+void log_pkt_skipped_to_buffer(PktHdr *pkt, PgSocket *client) {
```
Comment (felipethecarrot) [inline src/logging.c:219]: I think we can log the query as well, so we know what a long query looks like

**Review by felipethecarrot — CHANGES_REQUESTED** [review]

File: src/logging.c (hunk 4 above)
Comment (zainkabani) [reply src/logging.c:219]: I agree that we should be adding the queries to this but we currently don't have any way to use these in a useful way in the replayer.

I can create a separate PR for replayer and bouncer that uses these queries but for now the current changes get us a lot of value.

**Review by felipethecarrot — APPROVED** [review]

**Pull request merged**